great_expectations/_docs_decorators.py

```diff
@@ -60,6 +69,23 @@ def add(self, func: F) -> None:
             logger.exception(f"Could not add this function to the public API list: {func}")
             raise
 
+    def _add_to_class_registry(self, func: F) -> None:
+        if isinstance(func, type):
+            key = f"{func.__module__}.{func.__qualname__}"
+            self._class_registry[key].add(self.CLASS_DEFINITION)
+        else:
+            parts = func.__qualname__.split(".")
+            if len(parts) >= 2:  # noqa: PLR2004
```

Comment:
Could you add a reason for the noqa?

Also, this assumes that a dotted name will be a method. It could be a nested class or a class variable/constant? It seems like we could look at the type of func?